src/com.company/heap/KthSmallestInSortedMatrix.java

```diff
@@ -0,0 +1,44 @@
+package heap;
+
+import java.util.Comparator;
+import java.util.PriorityQueue;
+
+public class KthSmallestInSortedMatrix {
+    int findKthSmallestInSortedMatrix(int[][] matrix, int k) {
+        PriorityQueue<int[]> priorityQueue = new PriorityQueue<>(new Comparator<int[]>() {
+            @Override
+            public int compare(int[] o1, int[] o2) {
+                return Integer.compare(matrix[o1[0]][o1[1]], matrix[o2[0]][o2[1]]);
+            }
+        });
+
+        for (int i = 0; i < matrix[0].length; i++) {
```

Comment:
Why did you check the matrix[0] length? If the given matrix is not the square matrix, it will lead you to the wrong answer.